R/html_resources.R

```diff
@@ -216,6 +216,13 @@ discover_rmd_resources <- function(rmd_file, discover_single_resource) {
         output_render_files <- unlist(output_format[c(
           'includes', 'pandoc_args', 'logo', 'reference_doc', 'reference_docx', 'template'
         )])
+        # css needs to be copied for sass or bslib processing in formate rendering
+        if (is_bs_theme(resolve_theme(output_format[["theme"]]))) {
+          output_render_files <- c(output_render_files, unlist(output_format['css']))
+        } else if (any(needed <- needs_sass(output_format[['css']]))) {
+          output_render_files <- c(output_render_files, unlist(output_format[['css']][needed]))
+        }
```

Comment:
This is what I'd write, but it's up to you (I think your version is easier to read):

```suggestion
        needed <- if (is_bs_theme(resolve_theme(output_format[["theme"]]))) TRUE else {
          needs_sass(output_format[['css']])
        }
        output_render_files <- c(output_render_files, output_format[['css']][needed])
```